blog/_posts/2018-06-13-scala-213-collections.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: blog-detail
+post-type: blog
+by: Julien Richard-Foy
+title: Scala 2.13’s Collections
+---
+
+One more article about the standard collections, really? Indeed, during the last
+18 months a lot of work has been done on the collections and we’ve published
+several blog articles and given several talks to the explain various changes or
+challenges we were facing. This article attempts to summarize **what is going
+to change from an end-user perspective**.
+
+In case you’ve thoroughly followed our previous blog posts and talks, you might
+not learn much from this article. Otherwise, this is the perfect opportunity
+to catch up on the topic in a few minutes!
+
+The next section presents the internal changes in the collections implementation
+that might have some visible impact on the surface. Then, I will show why I think
+that the removal of `CanBuildFrom` made the API more beginner friendly. Next, I
+will introduce some new operations available in the collections. Finally, I
+will mention the main deprecations, the motivation behind them, and their
+recommended replacement.
+
+## Under The Hood: A Cleaner Ground
+
+![iceberg](/resources/img/blog/iceberg.jpeg)
+
+The most important change in the new collections framework is that transformation
+operations (such as `map` or `filter`) are now implemented in a way that works with both
+strict collections (such as `List`) and non-strict collections (such as `Stream`).
+This is a change because this was not the case before. Indeed, the previous
+implementations were strict and had to be overridden by non-strict collection types.
+You can find more details about that in
+[this blog post](/blog/2017/11/28/view-based-collections.html).
+
+The good news is that the new design is more **correct** in the sense that you can
+now implement custom non-strict collection types without having to worry about
```

Comment:
Is there some documentation somewhere with more details on this? Which collection traits are safe to extend for non-strict collections and which aren't? E.g. [`scala.collection.immutable.Set#concat`](https://github.com/scala/scala/blob/c5da7a0331e6549402253fb46053f1b0172ecac1/src/library/scala/collection/immutable/Set.scala#L48-L53) looks strict to me.

Reply:
Only traits whose name is prefixed with `StrictOptimized` are not safe to extend for non-strict collections. The code you’ve linked should be move to such a trait (it should not be in `SetOps` but in `StrictOptimizedSetOps`).